### DISCUSSION TITLE
How to integrate Yesvelte with TailwindCSS and DaisyUI? Problems with integrations.

### DISCUSSION THREAD

**Discussion by calionauta** [yesvelte/yesvelte#525] [Q&A]
####  When I installed Tailwind CSS and put the config in app.css (@tailwind base;@tailwind components;@tailwind utilities;), button background became transparent. Theme: tabler without other tailwind plugins.

**Before:**
![Screenshot 2023-06-19 09 16 38](https://github.com/yesvelte/yesvelte/assets/1719651/36d5a3fa-494d-4393-b2d6-a7af5fd1c887)

**After:**
![Screenshot 2023-06-19 09 16 19](https://github.com/yesvelte/yesvelte/assets/1719651/883e07d2-c144-4a5c-ba26-006aaa56f057)

##DaisyUI theme (not the component library)

#### NavbarItem with subitems (i.e. as Dropdown) didn't show the little arrow.  Theme: daisyUI without plugin DaisyUI component library.

**Tabler theme:**
![Screenshot 2023-06-19 09 24 48](https://github.com/yesvelte/yesvelte/assets/1719651/53f3fd8c-4ae9-44b2-8f93-1aff1dcab0d7)

**Daisyu theme:**
![Screenshot 2023-06-19 09 23 39](https://github.com/yesvelte/yesvelte/assets/1719651/4da6386f-47f0-42b8-8cd7-2cafdc0d3dc2)

## DaisyUI component library
1. When I installed the component library DaisyUI (not only set the daisyui theme of Yesvelte), some components didn't work as expected. 
 
####  1.1. The footer of the page turned dark. Theme: tabler with plugin DaisyUI component library.

**Before plugin:**
![Screenshot 2023-06-19 09 20 10](https://github.com/yesvelte/yesvelte/assets/1719651/8d88ad87-fce4-4cf5-949c-68f62e76b55a)

**After plugin:**
![Screenshot 2023-06-19 09 19 12](https://github.com/yesvelte/yesvelte/assets/1719651/03d44cf9-3f50-4c0c-b05e-e51ee41c06cf)

#### 1.2. FormAutocomplete with multiple items show some selected items outside its area and border. Theme: daisyUI with plugin DaisyUI component library.

**Before plugin:**
![Screenshot 2023-06-19 09 21 22](https://github.com/yesvelte/yesvelte/assets/1719651/f16060e2-e29f-4ad3-b552-60a88ad12f43)

**After plugin:**
![Screenshot 2023-06-19 09 22 53](https://github.com/yesvelte/yesvelte/assets/1719651/6036f0fb-ab47-4cf1-ba03-7283fcc9dd75)

#### 1.3. Elements that were fitting in a single line within their columns are now breaking lines, possibly due to the size of the components with the DaisyUI configuration.

**Before plugin:**
![Screenshot 2023-06-19 09 24 43](https://github.com/yesvelte/yesvelte/assets/1719651/141f4c7a-930a-40b2-bb20-aefaac474c32)

**After plugin:**
![Screenshot 2023-06-19 09 23 45](https://github.com/yesvelte/yesvelte/assets/1719651/f2bea5ac-32f7-43c3-966d-4c2290b8b50a)

**Comment by calionauta** [comment]
Theme daisyui of yesvelte without DaisyUI plugin: The icons are outside the input field.

![Screenshot 2023-06-19 10 16 02](https://github.com/yesvelte/yesvelte/assets/1719651/e04ac64a-b2f0-4e55-aca8-37f331033800)

**Comment by TheHadiAhmadi** [comment]
thanks for details, these components are independent from tailwind or daisyui, we change css classes and it is not possible to use `daisyui` package with our components.

for example for button component, daisyui has a `btn` class, we change it to `.y-button` using sass's `@extend` (similar to tailwind's `@apply`) and remove original `btn` class. 

if you need tailwind in your project (YeSvelte's daisyui theme doesn't depend on tailwind), you can import 'app.css' before 'yesvelte/css/daisyui.min.css'. (this should fix first button color problem)

for NavbarItem icon, daisyui theme of yesvelte doesn't support Arrow down icon. https://www.yesvelte.com/docs/navbar#Dropdown, it should be easy to implement using css.

**Reply by calionauta** [reply]
thank you

**Reply by TheHadiAhmadi** [reply]
You're welcome.

Our main priority is with tabler theme, and daisyui theme have some style related problems that should be fixed.

Good luck
